### PR TITLE
Add warnings on AWS credentials errors.

### PR DESCRIPTION
**Pull request by fmassot**

### CODE REVIEW

File: quickwit/quickwit-aws/src/lib.rs [hunk 1]
```diff
@@ -98,11 +99,15 @@ impl ExtendedChainProvider {
 #[async_trait]
 impl ProvideAwsCredentials for ExtendedChainProvider {
     async fn credentials(&self) -> Result<AwsCredentials, CredentialsError> {
-        if let Ok(credentials) = self.web_identity_provider.credentials().await {
-            return Ok(credentials);
+        match self.web_identity_provider.credentials().await {
+            Ok(credentials) => return Ok(credentials),
+            Err(error) => {
+                warn!(err=?error, "Failed to find AWS credentials from `WebIdentityProvider`.")
```
Comment (guilload) [inline quickwit/quickwit-aws/src/lib.rs:105]: We will always display this warning for users who rely on the "regular" `ChainProvider`. Let's store the error from the first provider and display it only if both authentication methods fail. We can also add debug statements to display which method was elected.

File: quickwit/quickwit-aws/src/lib.rs [hunk 2]
```diff
@@ -98,12 +99,21 @@ impl ExtendedChainProvider {
...
+        let chain_provider = self.chain_provider.credentials().await;
+
+        match (identity_provider, chain_provider) {
+            (Ok(credentials), _) => return Ok(credentials),
+            (_, Ok(credentials)) => return Ok(credentials),
+            (Err(identity_error), Err(chain_error)) => {
+                error!(
+                    web_identity_error=?identity_error,
+                    chain_provider_error=?chain_error,
+                    "Failed to find AWS credentials from `WebIdentityProvider` or `ChainProvider`.",
```
Comment (fmassot) [inline quickwit/quickwit-aws/src/lib.rs:112]:
```suggestion
                    "Failed to find AWS credentials from `WebIdentityProvider` and `ChainProvider`.",
```

File: quickwit/quickwit-aws/src/lib.rs [hunk 3]
```diff
@@ -98,12 +99,21 @@ impl ExtendedChainProvider {
 #[async_trait]
 impl ProvideAwsCredentials for ExtendedChainProvider {
     async fn credentials(&self) -> Result<AwsCredentials, CredentialsError> {
-        if let Ok(credentials) = self.web_identity_provider.credentials().await {
-            return Ok(credentials);
-        }
-        if let Ok(credentials) = self.chain_provider.credentials().await {
-            return Ok(credentials);
+        let identity_provider = self.web_identity_provider.credentials().await;
+        let chain_provider = self.chain_provider.credentials().await;
```
Comment (guilload) [inline quickwit/quickwit-aws/src/lib.rs:103]: Why resolve the second one if the first one succeeded?

```rust
let identity_provider_error = match self.web_identity_provider.credentials().await {
  Ok(credentials) => return Ok(credentials),
  Err(error) => error,
};
let chain_provider_error = match self.chain_provider.credentials().await {
  Ok(credentials) => return Ok(credentials),
  Err(error) => error,
};

error!(...)
Err(...)
```

Comment (fmassot) [reply quickwit/quickwit-aws/src/lib.rs:103]: nice, this is better indeed

**Comment by fmassot** [comment]
Closing as this should become obsolete with #3168

**Pull request closed without merging**